### PR TITLE
[Merged by Bors] - chore(CategoryTheory): generalize universes for multiequalizers

### PR DESCRIPTION
**Pull request by joelriou**
Multi(co)equalizers involve two index sets. Until now, they were assumed to be in the same universe. In this PR, we allow two distinct universes. This shall ease the formalisation of (co)ends, which require heterogeneous universes.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by dagurtomas** [comment]
Thanks!

maintainer merge

File: Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean [hunk 1]
```diff
@@ -151,9 +152,10 @@ structure MulticospanIndex (C : Type u) [Category.{v} C] where
 
 /-- This is a structure encapsulating the data necessary to define a `Multispan`. -/
 -- Porting note(#5171): linter not ported yet
```
Comment (TwoFX) [inline Mathlib/CategoryTheory/Limits/Shapes/Multiequalizer.lean:154]:
```suggestion
-- Porting note(#5171): has_nonempty_instance linter not ported yet
```

**Review by TwoFX — APPROVED** [review]
Thanks!
bors d+

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**